cosmos/client.go

```diff
@@ -83,7 +81,10 @@ func (c *Client) BuildAndBroadcastMsg(msg sdktypes.Msg, accName, accPassword str
 		if !ok {
 			return nil, errors.New("result data is not the right type")
 		}
-		return &data.TxResult, nil
+		if data.TxResult.Result.IsErr() {
```

Comment:
What do you think about replacing waiting for event to that [halper func](https://github.com/tendermint/tendermint/blob/b73cfe878682e0b73d9a21ec1d8dc456ddd90215/rpc/client/helpers.go#L58). I just found it and it does exactly the same steps as BuildAndBroadcastMsg needs

Reply:
The helper function was almost perfect but it's for event and not transaction.
But I switch to `EventQueryTxFor(encodedTx).String()` to construct the query instead of doing it manually.